pages/api/navigation.ts

```diff
@@ -0,0 +1,51 @@
+import qs from "qs";
+import { getStrapiURL } from "../../lib/api";
+import { transformLinks } from "../../utils";
+
+const navigationLinks = async (req, res) => {
+  try {
+    const { path = "/oeth-nav-links", options = {}, params = {} } = req.body;
+
+    // Build request URL
+    const queryString = qs.stringify(params);
+
+    const requestUrl = `${getStrapiURL(
+      `/api${path}${queryString ? `?${queryString}` : ""}`
+    )}`;
+
+    // Trigger API call
+    const result = await fetch(requestUrl, {
+      headers: {
+        method: "GET",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${process.env.STRAPI_API_KEY}`,
+      },
+      ...options,
+    }).then((res) => res.json());
+
+    const navLinks = transformLinks(result?.data || []);
+
+    return res.json({
+      data: navLinks,
+    });
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({
+      error: error.message,
+      data: [],
+    });
+  }
+};
+
+const handler = async (req, res) => {
+  const { method } = req;
+  switch (method) {
+    case "POST":
```

Comment:
nit: Why `POST` instead of `GET`?

Reply:
You are right this should be GET. Thank you, odd that it works either way though. I wonder if this is actually checking the http headers to allow